ivy/functional/frontends/torch/spectral_ops.py

```diff
@@ -0,0 +1,15 @@
+import ivy
+from ivy.functional.frontends.torch.func_wrapper import (
+    to_ivy_arrays_and_back,
+)
+from ivy.func_wrapper import with_supported_dtypes
+
+
+@with_supported_dtypes({"2.0.1 and below": "int"}, "torch")
+@to_ivy_arrays_and_back
+def hann_window(
+    window_length, periodic=True, *, dtype=None, requires_grad=False, out=None
```

Comment:
The parameters for `hann_window` torch frontend should be the same as torch's as shown [here](https://pytorch.org/docs/stable/generated/torch.hann_window.html). It seems to be missing `layout` and `device` kwarg

Reply:
Yes, that what I have implemented initially but it fails test case..

Reply:
> The parameters for `hann_window` torch frontend should be the same as torch's as shown [here](https://pytorch.org/docs/stable/generated/torch.hann_window.html). It seems to be missing `layout` and `device` kwarg. It also should not have `out` parameter

<img width="758" alt="image" src="https://github.com/unifyai/ivy/assets/88357044/f621add8-e0ac-421e-ab11-89623b77922f">

Reply:
The `device` and `layout` kwarg should be in the frontend parameter but not in the `ivy.hann_window` kwargs as `ivy.hann_window` does not support them.

Reply:
you mean, Implement `device` and `layout` kwarg  in function paramter but dont want  to use it return statement ivy.hann_window() am i right ?

Reply:
Right. For `layout` you might want to try `ivy.strides()` and `ivy.as_strided()`. For `device` you might want to try `ivy.to_device()`.

Reply:
ivy.hann_windows expects floating dtype...!